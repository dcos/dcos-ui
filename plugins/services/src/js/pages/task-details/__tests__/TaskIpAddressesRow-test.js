jest.mock("#SRC/js/stores/DCOSStore");

const TaskIpAddressesRow = require("../TaskIpAddressesRow").default;
const renderer = require("react-test-renderer");
const React = require("react");
const DCOSStore = require("#SRC/js/stores/DCOSStore");
const Application = require("#PLUGINS/services/src/js/structs/Application");
const Pod = require("#PLUGINS/services/src/js/structs/Pod");

describe("TaskIpAddressesRow", function() {
  describe("with an Application", function() {
    const serviceMock = new Application({
      tasks: [{ id: "foo", ipAddresses: [{ ipAddress: "127.0.0.1" }] }]
    });

    beforeEach(function() {
      DCOSStore.serviceTree = {
        getServiceFromTaskID() {
          return serviceMock;
        }
      };
    });

    it("renders IP Addresses", function() {
      const taskId = "foo";

      const thisInstance = renderer.create(
        <TaskIpAddressesRow taskId={taskId} />
      );

      const tree = thisInstance.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("with a Pod", function() {
    const serviceMock = new Pod({
      instances: [{ id: "foo.bar", networks: [{ addresses: ["0.0.0.0"] }] }]
    });

    beforeEach(function() {
      DCOSStore.serviceTree = {
        getServiceFromTaskID() {
          return serviceMock;
        }
      };
    });

    it("renders IP Addresses", function() {
      const taskId = "foo.bar.container-1";

      const thisInstance = renderer.create(
        <TaskIpAddressesRow taskId={taskId} />
      );

      const tree = thisInstance.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
