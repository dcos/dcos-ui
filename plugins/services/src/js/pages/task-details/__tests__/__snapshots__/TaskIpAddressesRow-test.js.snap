// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskIpAddressesRow with a Pod renders IP Addresses 1`] = `
<div
  className="configuration-map-row table-row"
>
  <div
    className="configuration-map-label"
  >
    <span
      className={undefined}
    >
      IP Addresses
    </span>
  </div>
  <div
    className="configuration-map-value"
  >
    0.0.0.0
  </div>
</div>
`;

exports[`TaskIpAddressesRow with an Application renders IP Addresses 1`] = `
<div
  className="configuration-map-row table-row"
>
  <div
    className="configuration-map-label"
  >
    <span
      className={undefined}
    >
      IP Addresses
    </span>
  </div>
  <div
    className="configuration-map-value"
  >
    127.0.0.1
  </div>
</div>
`;
